File: src/actor_tf2_broadcaster.py
#!/usr/bin/env python  
import rospy

# Because of transformations
import tf_conversions
import tf2_ros
import geometry_msgs.msg
import actor.msg


def handle_actor_pose(msg, actor):
    br = tf2_ros.TransformBroadcaster()
    t = geometry_msgs.msg.TransformStamped()

    t.header.stamp = rospy.Time.now()
    t.header.frame_id = "world"
    t.child_frame_id = actor
    t.transform.translation.x = msg.x
    t.transform.translation.y = msg.y
    t.transform.translation.z = 0.0
    q = tf_conversions.transformations.quaternion_from_euler(0, 0, msg.theta)
    t.transform.rotation.x = q[0]
    t.transform.rotation.y = q[1]
    t.transform.rotation.z = q[2]
    t.transform.rotation.w = q[3]

    br.sendTransform(t)

if __name__ == '__main__':
    rospy.init_node('tf2_actor_broadcaster')
    turtlename = rospy.get_param('~actor')
    rospy.Subscriber('/%s/pose' % actor,
                     actor.msg.Pose,
                     handle_actor_pose,
                     actor)
    rospy.spin()

File: src/actor_tf2_broadcaster.cpp
#include <ros/ros.h>
#include <geometry_msgs/TransformStamped.h>
#include <geometry_msgs/PoseStamped.h>
#include <gazebo_msgs/ModelStates.h> 
#include <geometry_msgs/Pose.h>
#include <tf2_ros/transform_broadcaster.h>
#include <nav_msgs/Odometry.h>
#include <iostream>
#include <tf2/LinearMath/Quaternion.h>
#include <tf2_geometry_msgs/tf2_geometry_msgs.h>

geometry_msgs::PoseStamped actorstate;

void model_cb(const gazebo_msgs::ModelStates::ConstPtr& msg)
{
  gazebo_msgs::ModelStates current_states = *msg;
  int actorArrPos = 999;

  //search for the drone
  for (int i=0; i< current_states.name.size(); i++)
  {
  	if(current_states.name[i] == "actor")
  	{
  		actorArrPos = i;
  		break;
  	}
  	
  }
  if (actorArrPos == 999)
  {
  	std::cout << "actor is not in world" << std::endl;
  }else{
  	//assign actor's pose to pose stamped
  	actorstate.pose = current_states.pose[actorArrPos];
  	actorstate.header.stamp = ros::Time::now();
    std::cout <<  actorstate << std::endl;

    //do transformed stamped stuff
    transformStamped.header.stamp = ros::Time::now();
  	transformStamped.header.frame_id = "world";
  	// not sure if correct child frame...maybe it's "head" of the actor?
  	transformStamped.child_frame_id = actor;
 	transformStamped.transform.translation.x = msg->x;
  	transformStamped.transform.translation.y = msg->y;
  	transformStamped.transform.translation.z = 0.0;
  	tf2::Quaternion q;
  	q.setRPY(0, 0, msg->theta);
  	transformStamped.transform.rotation.x = q.x();
  	transformStamped.transform.rotation.y = q.y();
  	transformStamped.transform.rotation.z = q.z();
  	transformStamped.transform.rotation.w = q.w();

  	br.sendTransform(transformStamped);

}
int main(int argc, char **argv) {
	ros::init(argc, argv, "acotr_tf2_broadcaster");
	ros::NodeHandle n;
	ros::Subscriber currentHeading = n.subscribe<gazebo_msgs::ModelStates>("/gazebo/model_states", 10, model_cb);
	ros::Rate rate(20.0);

	ros::spin()
	return 0;
};